Feature/firestore callbacks

rmacgill opened the pull request.

rmacgill committed:
Works for Current Request

- Proper Firestore callbacks seemt to be working for the current request. No longer query Firestore every 3 seconds.

rmacgill committed:
Driver Open Requests on Callback

- Driver users no longer poll Firestore for pending ride requests
- Hooked up callbacks to the ride request collection, so the driver view refreshes when something changes in the database

rmacgill committed:
Update DriverMarkerInfoDialog.java

- Fixed a toast bug that could occur if a ride was cancelled or accepted before a driver viewing the details hit accept

rmacgill committed:
Update RideRequestTests.java

-Removed test that now relies on asynchronous functionality

dijonron reviewed (commented):
Bug: If driver has accepts ride and signs out, they are given the rejected ride notification

rmacgill committed:
Bugfix

- Fixed the erroneous request status change notification
- Added a way to completely deconstruct the cache so that it's safer to sign out and switch users
